pkg/deployment/depresource/cancel_plan.go

```diff
@@ -51,8 +50,8 @@ func (params CancelPlanParams) Validate() error {
 		merr = multierror.Append(merr, deputil.NewInvalidDeploymentIDError(params.DeploymentID))
 	}
 
-	if params.Type == "" {
-		merr = multierror.Append(merr, errors.New("deployment resource type cannot be empty"))
+	if !slice.HasString(util.ValidTypes, params.Type) {
```

Comment:
I had a chat with @marclop  the week before and we agreed not to do any validation of the accepted types for simplicity. You were not supposed to know this @karencfv but let's be consistent and move this validation